### PR TITLE
[CHK-2230] feat(jwt token utils): use commons jwt token utils

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Update transaction service to use commons jwt token utils 

#### List of Changes

1. use jwt token utils from commons. 
2. add different key for npg notification uri jwt token
3. rename the key for ecommerce

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -62,7 +62,9 @@ transactions.ecommerce.retry.offset=${TRANSACTIONS_RETRY_OFFSET}
 
 transactions.closure_handler.retry_interval=${CLOSURE_RETRY_INTERVAL}
 
-jwt.secret=${JWT_SECRET}
+jwt.ecommerce.secretKey=${JWT_ECOMMERCE_SECRET}
+npg.notification.jwt.secret=${JWT_NPG_NOTIFICATION_SECRET}
```
Comment (ciuffagianluca) [inline src/main/resources/application.properties:66]: @Fernando-Granato is thi the right name for our variable ?

File: .devops/code-review-pipelines.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ resources:
     - repository: pagopaEcommerceLocal
       type: github
       name: pagopa/pagopa-ecommerce-local 
-      ref: main
+      ref: CHK-2241-update-env
```
Comment (ciuffagianluca) [inline .devops/code-review-pipelines.yml:15]: to be updated

File: helm/values-dev.yaml [hunk 3]
```diff
@@ -132,21 +132,23 @@ microservice-chart:
...
+    JWT_NPG_NOTIFICATION_VALIDITY_TIME: "900"
   envSecret:
     MONGO_PASSWORD: mongo-ecommerce-password
     REDIS_PASSWORD: redis-ecommerce-password
     NODO_CONNECTION_STRING: nodo-connection-string
     NOTIFICATIONS_SERVICE_API_KEY: notifications-service-api-key
     ECOMMERCE_STORAGE_TRANSIENT_CONNECTION_STRING: ecommerce-storage-transient-connection-string
     PAYMENT_TRANSACTIONS_GATEWAY_API_KEY: payment-transactions-gateway-api-key
-    JWT_SECRET: sessions-jwt-secret
+    JWT_ECOMMERCE_SECRET: sessions-jwt-secret
```
Comment (Fernando-Granato) [inline helm/values-dev.yaml:144]: Add this variable to README file

File: helm/values-dev.yaml [hunk 4]
```diff
@@ -132,21 +132,23 @@ microservice-chart:
...
     PERSONAL_DATA_VAULT_API_KEY: personal-data-vault-api-key
     ECOMMERCE_PAYMENT_METHODS_APY_KEY: payment-method-api-key
     WALLET_API_KEY: wallet-api-key
     OTEL_EXPORTER_OTLP_HEADERS: elastic-otel-token-header
     NPG_API_KEY: npg-api-key
     NPG_CARDS_PSP_KEYS: npg-cards-psp-keys
+    JWT_NPG_NOTIFICATION_SECRET: npg-notification-signing-key
```
Comment (Fernando-Granato) [inline helm/values-dev.yaml:151]: Add this variable to README file

File: src/main/java/it/pagopa/transactions/commands/handlers/v1/TransactionActivateHandler.java [hunk 5]
```diff
@@ -208,8 +215,16 @@ public Mono<Tuple2<Mono<BaseTransactionEvent<?>>, String>> handle(
                         .collectList()
                         .flatMap(
                                 paymentRequestInfos -> jwtTokenUtils
-                                        .generateToken(transactionId, null)
+                                        .generateToken(
+                                                ecommerceSigningKey,
+                                                jwtEcommerceValidityTimeInSeconds,
+                                                new Claims(transactionId, null, null)
+                                        )
                                         .map(generatedToken -> Tuples.of(generatedToken, paymentRequestInfos))
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/transactions/commands/handlers/v1/TransactionActivateHandler.java:223]: Remove map and use only fold
```suggestion
                                      .fold(
                                                Mono::error,
                                                generatedToken -> Mono.just( Tuples.of(generatedToken, paymentRequestInfos)
                                        )
                                        
```

File: src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionActivateHandler.java [hunk 6]
```diff
@@ -209,8 +215,16 @@ public Mono<Tuple2<Mono<BaseTransactionEvent<?>>, String>> handle(
                         .collectList()
                         .flatMap(
                                 paymentRequestInfos -> jwtTokenUtils
-                                        .generateToken(transactionId, command.getData().orderId())
+                                        .generateToken(
+                                                ecommerceSigningKey,
+                                                jwtEcommerceValidityTimeInSeconds,
+                                                new Claims(transactionId, command.getData().orderId(), null)
+                                        )
                                         .map(generatedToken -> Tuples.of(generatedToken, paymentRequestInfos))
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionActivateHandler.java:223]: Remove map and use only fold
```suggestion
                                        
                                        .fold(
                                                Mono::error,
                                                generatedToken -> Mono.just( Tuples.of(generatedToken, paymentRequestInfos)
                                        )
```

**Review by Fernando-Granato — DISMISSED** [review]

**Review by infantesimone — APPROVED** [review]

**Review by Fernando-Granato — APPROVED** [review]

**Pull request merged**